astropy/table/tests/test_table.py

```diff
@@ -3592,3 +3592,10 @@ def test_table_replace_column_with_scalar(empty_table):
     # Direct replacement should never work.
     with pytest.raises(ValueError, match="cannot replace.*with a scalar"):
         t.replace_column("a", 2)
+
+
+@pytest.mark.parametrize("rows", [[], (), None])
```

Comment:
I tried an empty numpy array and it still triggers the bug.
```suggestion
@pytest.mark.parametrize("rows", [None, [], (), np.array([])])
```

Reply:
I threw in the tuple as a bonus. Original ask was only for empty list. Do people really go out of their way and pass in  empty numpy array?

Reply:
> Do people really go out of their way and pass in empty numpy array?

It could happen dynamically and make for a very confusing debug session for someone down the road.
I think we *have* to assume it'll happen somewhere, and make sure the API stays consistent through this change.